### PR TITLE
[Excel] (Custom Functions) Making web requests/etc their own topic

### PR DESCRIPTION
**Pull request by mscharlock**
This is just a separation out of the web request (XHR/Fetch) + streaming + websockets + cancelling functions from previous CFs docs.

### CODE REVIEW

**Comment by o365devx** [comment]
OPS Build status updates of commit _[b12d534](https://github.com/OfficeDev/office-js-docs-pr/pull/817/commits/b12d53485bda21c9544cebc99e990191cc6192bf)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-web-reqs.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-web-reqs.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C12%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201903121726091642-817%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=NnqQduS4VA1TQaY687Y%2FB6mwUn8frsAhOrD%2Fql65GnQ%3D&st=2019-03-12T17%3A22%3A27Z&se=2019-04-12T17%3A27%3A27Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/custom-functions-web-reqs.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+---
```
Comment (mscharlock) [inline docs/excel/custom-functions-web-reqs.md]: Note to self: check about cancellable handler invocations

File: docs/excel/custom-functions-web-reqs.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+ms.date: 03/11/2019
+description: Request, stream, and cancel streaming of external data to your workbook with custom functions in Excel
+title: Web requests and other data handling with custom functions (preview)
+localization_priority: Priority
+---
+
+# Receiving and handling data with custom functions
+
+One of the ways that custom functions enhance Excel's power is by receiving data from locations other than the workbook, such as the web or a server (through WebSockets). Custom functions can request data through XHR and fetch reqeuests as well as stream this data in real time.
```
Comment (davidchesnut) [inline docs/excel/custom-functions-web-reqs.md]:
```suggestion
One of the ways that custom functions enhance Excel's power is by receiving data from locations other than the workbook, such as the web or a server (through WebSockets). Custom functions can request data through XHR and fetch requests as well as stream this data in real time.
```

File: docs/excel/custom-functions-web-reqs.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+You can request external data through an API like [`Fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) or by using `XmlHttpRequest` [(XHR)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), a standard web API that issues HTTP requests to interact with servers.
+
+Within custom functions runtime, XHR implements additional security measures by requiring [Same Origin Policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) and simple [CORS](https://www.w3.org/TR/cors/).
+
+Note that a simple CORS implementation cannot use cookies and only supports simple methods (GET, HEAD, POST). Simple CORS accepts simple headers with field names `Accept`, `Accept-Language`, `Content-Language`. You can also use a Content-Type header in simple CORS, provided that the content type is `application/x-www-form-urlencoded`, `text/plain`, or `multipart/form-data`.
+
+### XHR example
+
+In the following code sample, the getTemperature function calls the sendWebRequest function to get the temperature of a particular area based on thermometer ID. The sendWebRequest function uses XHR to issue a GET request to an endpoint that can provide the data.
```
Comment (davidchesnut) [inline docs/excel/custom-functions-web-reqs.md]: Be sure to bold function names throughout like **getTemperature**

**Comment by o365devx** [comment]
OPS Build status updates of commit _[ebcc885](https://github.com/OfficeDev/office-js-docs-pr/pull/817/commits/ebcc885de6ad51d8edc97f1d08eb1899b128cf96)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-web-reqs.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-web-reqs.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C21%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201903211633016302-817%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=y5auDLng8afgBgtIw%2BfbMuiRZdhkh0AQ9HkNjWemLPI%3D&st=2019-03-21T16%3A29%3A25Z&se=2019-04-21T16%3A34%3A25Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
OPS Build status updates of commit _[37fc8a9](https://github.com/OfficeDev/office-js-docs-pr/pull/817/commits/37fc8a9ef658eb1d3639904be6271d4c7383ac1f)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-json.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-json.md) | :warning:Warning | | [Details](#user-content-docs%2Fexcel%2Fcustom-functions-json.md)
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-web-reqs.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-web-reqs.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/toc.yml) | :white_check_mark:Succeeded | |

<a name="docs%2Fexcel%2Fcustom-functions-json.md"></a>
### [docs/excel/custom-functions-json.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-json.md)
 - **Line 124**: **[Warning]** Illegal link: `&lt;a href=&quot;excel/custom-functions-overview.md#canceling-a-function&quot;&gt;Canceling a function&lt;/a&gt;` -- missing bookmark. The file excel/custom-functions-overview.md doesn&#39;t contain a bookmark named canceling-a-function.
 - **Line 125**: **[Warning]** Illegal link: `&lt;a href=&quot;excel/custom-functions-overview.md#streaming-functions&quot;&gt;Streaming functions&lt;/a&gt;` -- missing bookmark. The file excel/custom-functions-overview.md doesn&#39;t contain a bookmark named streaming-functions.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C21%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201903211640352761-817%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=O8XZYkOhbLbuPhg4%2BYWZSMKqRh%2B5Y0Qf2rvwqjl88J0%3D&st=2019-03-21T16%3A36%3A57Z&se=2019-04-21T16%3A41%3A57Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
OPS Build status updates of commit _[9f720e2](https://github.com/OfficeDev/office-js-docs-pr/pull/817/commits/9f720e2a00dd59c347af8d52fe4244ff9773bb7c)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-json.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-json.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-web-reqs.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/excel/custom-functions-web-reqs.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-xhrstream/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C21%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201903211722071217-817%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=lWuKCSJmE04nBET7GvhdNMp9Yw2XaCmEmHrmcucya%2Fc%3D&st=2019-03-21T17%3A18%3A31Z&se=2019-04-21T17%3A23%3A31Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**